app/src/main/java/tech/pauly/findapet/discover/AnimalDetailsViewModel.java

```diff
@@ -0,0 +1,112 @@
+package tech.pauly.findapet.discover;
+
+import android.databinding.ObservableBoolean;
+import android.databinding.ObservableField;
+import android.databinding.ObservableInt;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import tech.pauly.findapet.R;
+import tech.pauly.findapet.data.models.Animal;
+import tech.pauly.findapet.data.models.Media;
+import tech.pauly.findapet.data.models.Option;
+import tech.pauly.findapet.data.models.Photo;
+import tech.pauly.findapet.data.models.PhotoSize;
+import tech.pauly.findapet.shared.AnimalDetailsUseCase;
+import tech.pauly.findapet.shared.BaseViewModel;
+import tech.pauly.findapet.shared.ResourceProvider;
+import tech.pauly.findapet.shared.TransientDataStore;
+
+public class AnimalDetailsViewModel extends BaseViewModel {
+
+    private final ResourceProvider resourceProvider;
+    public ObservableField<String> name = new ObservableField<>("");
+    public ObservableField<String> imageUrl = new ObservableField<>("");
+    public ObservableField<String> ageType = new ObservableField<>("");
+    public ObservableField<String> breeds = new ObservableField<>("");
+    public ObservableInt sex = new ObservableInt(R.string.missing);
+    public ObservableInt size = new ObservableInt(R.string.missing);
+    public ObservableField<String> options = new ObservableField<>("");
+    public ObservableField<String> description = new ObservableField<>("");
+    public ObservableBoolean descriptionVisibility = new ObservableBoolean(false);
+    public ObservableBoolean optionsVisibility = new ObservableBoolean(false);
+
+    private AnimalDetailsViewPagerAdapter viewPagerAdapter;
+
+    @Inject
+    AnimalDetailsViewModel(TransientDataStore dataStore,
+                           AnimalDetailsViewPagerAdapter viewPagerAdapter,
+                           ResourceProvider resourceProvider) {
+        this.viewPagerAdapter = viewPagerAdapter;
+        this.resourceProvider = resourceProvider;
+        this.viewPagerAdapter.setViewModel(this);
+        AnimalDetailsUseCase useCase = dataStore.get(AnimalDetailsUseCase.class);
+        if (useCase != null) {
+            Animal animal = useCase.getAnimal();
+            name.set(animal.getName());
+            sex.set(animal.getSex().getFormattedName());
+            size.set(animal.getSize().getFormattedName());
+            ageType.set(animal.getAge() + " " + resourceProvider.getString(animal.getType().getSingularName()));
```

Comment:
Going back and forth on where this stuff belongs. Most of it is unique to this class, but some can be genericized, maybe a util/formatter.